R/zzz.R

```diff
@@ -0,0 +1,24 @@
+
+#' Register engines to support Quarto vignettes
+#' @importFrom tools vignetteEngine
+#' @noRd
+.onLoad <- function(libname, pkgname) { # args ignored
+  vignetteEngine(name = "pdf",
+                 package = "quarto",
+                 pattern = "[.]qmd$",
+                 weave = function(file, ..., encoding = "UTF-8") {
+                   quarto_render(file, ..., output_format = "pdf")
+                 },
+                 tangle = vignetteEngine("knitr::rmarkdown")$tangle,
+                 aspell = vignetteEngine("knitr::rmarkdown")$aspell
+                 )
```

Comment:
> Do you have experience on CRAN to build LaTeX vignette ?

Not with Quarto, but certainly with Rmd.  This vignette for my DTAT package serves to demonstrate the reproducibility of calculations for a peer-reviewed paper, e.g.:
https://cran.r-project.org/web/packages/DTAT/vignettes/dtat.pdf

I would tend to discourage gratuitous [FUD](https://en.wikipedia.org/wiki/Fear,_uncertainty,_and_doubt) here about building LaTeX stuff on CRAN.  I don't recall myself ever having to 'downgrade' a PDF vignette to HTML for any technical reason.

Reply:
Sorry if my tone was not the good one. (sometimes non native english speaker catch me ... 😞) 
I didn't mean to sound like I was doubtful about it working well. It was mainly question about what can really be done considering how quarto is working by default with [quarto pdf engine](https://quarto.org/docs/output-formats/pdf-engine.html#quarto-pdf-engine)

`format: pdf` require LaTeX and some packages. Pandoc's template and some quarto feature have requirement about LaTeX packages. By default, quarto will try to install missing LaTeX packages when rendering errors are found, and then retrigger a render. By "LaTeX installation", I mean which LaTeX extension are available or not. I don't think it is seen as good practice to automatically update tlmgr and install new missing LaTeX package. So I meant it was safer to opt-out this feature, or use PDF output produced by `format: typst` which is quicker to render. 

I don't know the list of available LaTeX component from CTAN available on CRAN build machine to compare with Pandoc's and Quarto requirement. Maybe this is full texlive install available and so no problem ever. 

However, as pointed out in the manual you rightly referenced, including the output in the source would make sure to LaTeX extension installation are needed. So that is a way to make sure it would work. 

My concerns is preventing any problem with package building on CRAN because of how quarto works, and what impact building vignette with quarto could have. Maybe this is nothing to worry about. 

Anyhow, I hope this clarifies my initial thoughts.